### ISSUE TITLE
When training the GAN model, how many epochs  do we need to train

### ISSUE THREAD

**Issue by xuhaohao86** [junyanz/pytorch-CycleGAN-and-pix2pix#1127]
Hello,
In your article, Cyclegan is trained a total of 200 epochs.

But when I read the latest GAN article-"Reusing Discriminators for Encoding: Towards Unsupervised Image-to-Image Translation",  the author compares some gan models, including Cyclegan. In the article, All models are trained over 100K iterations.

so,i want to konw , When training the GAN model,  how many epochs  do we need to train

**Comment by junyanz** [comment]
There is no optimal training epochs/iterations for GANs training (or for deep learning models in general). The number depends on your dataset size, data type, and application. Authors often use the same epochs/iterations for a relatively fair comparison. You can save the checkpoints frequently and evaluate all the saved checkpoints afterward.

**Comment by xuhaohao86** [comment]
ok,Thanks

**Comment by Tyotw** [comment]
Hello，I am a new guy ,so I want to ask a stupid question: **Where can I set the epoch times** as I want , Thanks.

**Comment by junyanz** [comment]
`--n_epochs` and `--n_epochs_deplay`. See this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/options/train_options.py#L30) for more details.

**Comment by SirRacha** [comment]
@junyanz 
I set --n_epochs to 20, but it did not stop.   How can I get it to stop training? Is there another parameter?  Also, how can I only use checkpoint 20 for testing. Thanks!

**Comment by jysa01** [comment]
Check line 30 in train_options.py: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/f13aab8148bd5f15b9eb47b690496df8dadbab0c/options/train_options.py#L30
Setting `--n_epochs` to 20 will train for 20 epochs with the initial learning rate and training continues for `n_epochs_decay`. You may stop the training at 20 epochs by distributing epochs between the n_epochs and n_epochs_decay (say 10 and 10 ?)as per your requirement. 


Check line 53 in the base_options.py :  https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/f13aab8148bd5f15b9eb47b690496df8dadbab0c/options/base_options.py#L53

If you have saved the 20th model during training, you should be able to load it when you test either by passing` --epoch 20 `with your test command.
